clippy_lints/src/manual_assert.rs

```diff
@@ -62,25 +63,22 @@ impl<'tcx> LateLintPass<'tcx> for ManualAssert {
             };
             let cond_sugg = sugg::Sugg::hir_with_applicability(cx, cond, "..", &mut applicability).maybe_par();
             let semicolon = if is_parent_stmt(cx, expr.hir_id) { ";" } else { "" };
-            let sugg = format!("assert!({not}{cond_sugg}, {format_args_snip}){semicolon}");
-            // we show to the user the suggestion without the comments, but when applying the fix, include the
-            // comments in the block
+            let base_sugg = format!("assert!({not}{cond_sugg}, {format_args_snip}){semicolon}");
+
+            let indent = indent_of(cx, expr.span);
+            let full_sugg = reindent_multiline(format!("{comments}{base_sugg}").into(), true, indent);
```

Comment:
While we're already here and have to change something anyway: since this `base_sugg` is only used once we could inline it into the format macro directly to get rid of an allocation
```suggestion
            let indent = indent_of(cx, expr.span);
            let full_sugg = reindent_multiline(&*format!("{comments}assert!({not}{cond_sugg}, {format_args_snip}){semicolon}"), true, indent);
```
(may need some formatting changes if the line is too long)

-----

Also, can you move all of this code that prepares/formats the suggestion into the `span_lint_and_then` closure? The lint is in pedantic, so often the lint isn't enabled and we don't need to emit/prepare any suggestion as the closure won't be called.